src/_different.ts

```diff
@@ -0,0 +1,16 @@
+// import * as _ from 'lodash';
+
+// console.log(_.difference([2, 3,4], [2, 1]));
+
+const _difference = <T>(array: T[], conditionArray: T[]): T[] => {
+    const newArr: T[] = [];
+    for (let item of array) {
+        !conditionArray.includes(item) && newArr.push(item);
```

Comment:
method `includes` có độ phức tạp `O(n)` (bên trong nó thực chất là 1 vòng lặp để tìm ra phần tử).
Thuật toán của em có độ phức tạp là `O(n^2)` do 2 vòng lặp lồng nhau.
Bài toán này có thể xử lý tốt hơn bằng cách chuyển `conditionArray` thành `Set` rồi dùng `Set.has` để kiểm tra thay vì `Array.includes`. Như vậy độ phức tạp sẽ còn `O(n)`

![image](https://user-images.githubusercontent.com/117837318/201106527-f9b4148f-4b15-4fa3-b8e4-f23fa62b635c.png)